dds/DCPS/Service_Participant.cpp

```diff
@@ -253,9 +253,15 @@ Service_Participant::reactor_owner() const
 void
 Service_Participant::shutdown()
 {
+  // When we are already shutdown just let the shutdown be a noop
+  if (shut_down_) {
+    return;
+  }
+
   shut_down_ = true;
   try {
     TransportRegistry::instance()->release();
+    TransportRegistry::instance()->close();
```

Comment:
close() is declared static so invoke it as `TransportRegistry::close()`.  Also does close() need to be a distinct operation from release()?